File: samples/spring-mvc-boot-3/src/main/java/io/openapiprocessor/samples/ApiController.java
package io.openapiprocessor.samples;

import io.openapiprocessor.openapi.api.BarApi;
import io.openapiprocessor.openapi.api.FooApi;
import io.openapiprocessor.openapi.model.Foo;
import org.springframework.web.bind.annotation.RestController;

import java.util.UUID;

@RestController
public class ApiController implements FooApi, BarApi {

    @Override
    public MappedBar postBar(MappedBar bar) {
        return bar != null ? bar : new MappedBar();
    }

    @Override
    public Foo postFoo(Foo foo) {
        return foo != null ? foo : new Foo("foo", UUID.randomUUID(), new MappedBar());
    }
}

File: samples/spring-mvc-boot-3-kt/src/main/kotlin/io/openapiprocessor/samples/ApiController.kt
package io.openapiprocessor.samples

import io.openapiprocessor.samples.api.BarApi
import io.openapiprocessor.samples.api.FooApi
import io.openapiprocessor.samples.model.Bar
import io.openapiprocessor.samples.model.Foo
import org.springframework.data.domain.Page
import org.springframework.data.domain.PageImpl
import org.springframework.data.domain.Pageable
import org.springframework.web.bind.annotation.RestController
import java.util.*

@RestController
class ApiController: BarApi, FooApi {

    override fun postFoo(foo: Foo?): Foo {
        return foo ?: Foo("foo", UUID.randomUUID(), Bar("value"))
    }

    override fun echoBar(bar: Bar): Bar {
        return bar
    }

    override fun getBars(pageable: Pageable?): Page<Bar> {
        return PageImpl(listOf(
            Bar("value 1"),
            Bar("value 2")
        ))
    }
}
